salt/client/__init__.py

```diff
@@ -1345,7 +1345,7 @@ def _prep_pub(self,
             payload_kwargs['kwargs'] = kwargs
 
         # If we have a salt user, add it to the payload
-        if self.opts['order_masters'] and 'user' in kwargs:
```

Comment:
I believe you want either order_masters or syndic_master, since they mean you are a syndic or a master-master.